basicswap/bin/prepare.py

```diff
@@ -1216,12 +1285,21 @@ def prepareDataDir(coin, settings, chain, particl_mnemonic, extra_opts={}):
             if coin == "monero":
                 if XMR_RPC_USER != "":
                     fp.write(f"rpc-login={XMR_RPC_USER}:{XMR_RPC_PWD}\n")
-            if coin == "wownero":
+            elif coin == "salvium":
+                if SAL_RPC_USER != "":
+                    fp.write(f"rpc-login={SAL_RPC_USER}:{SAL_RPC_PWD}\n")
+            elif coin == "wownero":
                 if WOW_RPC_USER != "":
                     fp.write(f"rpc-login={WOW_RPC_USER}:{WOW_RPC_PWD}\n")
-            if tor_control_password is not None:
-                for opt_line in monerod_proxy_config:
-                    fp.write(opt_line + "\n")
+
+            # Tor proxy for salvium or monero/wownero only
+            if coin in ("monero", "wownero", "salvium") and tor_control_password is not None:
```

Comment:
L1260 already gates this to these coins

```suggestion
            if tor_control_password is not None:
```